app/models/topic.rb

```diff
@@ -1040,7 +1040,9 @@ def invite(invited_by, username_or_email, group_ids = nil, custom_message = nil)
         raise UserExists.new(I18n.t("topic_invite.user_exists"))
       end
 
-      ensure_can_invite!(target_user, invited_by)
+      communication_preferences = UserCommunicationDefender.new(acting_user_id: invited_by.id, target_usernames: target_user.username).fetch_user_preferences
+      target_user_communication_preferences = communication_preferences.for_user(target_user.id)
+      raise NotAllowed.new(I18n.t("not_accepting_pms", username: target_user.username)) if !communication_preferences.acting_user_staff? && target_user_communication_preferences&.ignoring_or_muting?
```

Comment:
Design thought: Same question about the staff, maybe `ignoring_or_muting?` should be like this?

```ruby
def ignoring_or_muting?
  return false if acting_user.staff?
  is_muting || is_ignoring
end
```

Reply:
What about these staff checks then @tgxworld ? I guess they would then fall into the shorthand methods like `can_invite_user?` or `can_pm_user?`, that kind of thing?